MmSupervisorPkg/Drivers/MmPeiLaunchers/Common/MmIplCommon.c

```diff
@@ -397,8 +397,12 @@ QuerySupervisorVersion (
     ));
 
   if (VersionInfo->Version == 0) {
+ #ifdef __GNUC__

```

Comment:
Is it correct that we want this function to return EFI_SUCCESS if we are using the GCC combiler?  I'd assume we'd want it to still return EFI_SECURITY_VIOLATION.

Reply:
That will make GCC built supervisor never boot (which is part of the GCC building issues.) There is currently no effortless way to embed these versions into the final output.

At least for now, we would not check this on GCC built systems. But if others ever saw this message and have better ideas, please comment and welcome to contribute directly.

Reply:
Got it.  I assumed this was intended but just wanted to make sure.

Reply:
Right. This is only a workaround, so that it does not break the build all the time. But we will keep investigating with other methods, hoping to remove this branching logic some time.